### PR TITLE
build(deps): Bump android Appcompat to 1.4.1

### PR DESCRIPTION
**Pull request by gabrieldonadel**
## Summary

Currently we are using Appcompat in version 1.0.2 which is almost 4 years old now, this PR updates it to version 1.4.1.

Using Appcompat 1.0.2 was also causing a crash on RNTester due to an error where FontFamily's method was not found (Related to https://github.com/facebook/react-native/issues/33065)

Closes #31620

## Changelog

[Android] [Changed] - Bump android Appcompat to 1.4.1

## Test Plan

Use `./scripts/test-manual-e2e.sh` to test both RNTester and a new app

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/devsupport/RedBoxDialog.java [hunk 1]
```diff
@@ -323,7 +323,7 @@ public View getContentView() {
   }
 
   public boolean isShowing() {
-    return mDialog.isShowing();
+    return mDialog != null && mDialog.isShowing();
```
Comment (gabrieldonadel) [inline ReactAndroid/src/main/java/com/facebook/react/devsupport/RedBoxDialog.java:326]: RNTester was crashing without this extra check
image

**Comment by gabrieldonadel** [comment]
@cortinico I believe this commit https://github.com/facebook/react-native/commit/cd79317672e5c99636346f2abb641a688a4ceb82 broke CI, in particular the `test_android_template` workflow https://app.circleci.com/pipelines/github/facebook/react-native/12051/workflows/2592c9f6-217f-4595-9981-82f4097fecf3/jobs/234753

**Comment by gabrieldonadel** [comment]
Ohh never mind, I guess you already have a [PR](https://github.com/facebook/react-native/pull/33068) fixing this

**Comment by cortinico** [comment]
> Ohh never mind, I guess you already have a [PR](https://github.com/facebook/react-native/pull/33068) fixing this

Yup you're right. We're working on it, hopefully it will be out in the next hours. Sorry for the disruption 👍

**Comment by ShikaSD** [comment]
Hi, thanks  for upgrading it, do you mind updating the BUCK file for the dependency as well?

https://github.com/facebook/react-native/blob/369a7ab5d332be945137aaa487f0c09fc7fa8c57/ReactAndroid/src/main/third-party/android/androidx/BUCK#L538

**Comment by gabrieldonadel** [comment]
> Hi, thanks for upgrading it, do you mind updating the BUCK file for the dependency as well?
> 
> https://github.com/facebook/react-native/blob/369a7ab5d332be945137aaa487f0c09fc7fa8c57/ReactAndroid/src/main/third-party/android/androidx/BUCK#L538

Sure @ShikaSD, I can update that. My only question is regarding the `sha1` value, should I run some command that updates this automatically or do I need to get it from somewhere else?

**Comment by ShikaSD** [comment]
> Should I run some command that updates this automatically or do I need to get it from somewhere else?

It can be retrieved from google maven repo, I don't think we have any command that upgrades it automatically. If you struggle to find the hash, you can upgrade the version and then wait for the CI to fail, it should have a new hash there.

**Comment by gabrieldonadel** [comment]
> It can be retrieved from google maven repo, I don't think we have any command that upgrades it automatically. If you struggle to find the hash, you can upgrade the version and then wait for the CI to fail, it should have a new hash there.

Ohh got it! Thanks @ShikaSD

**Comment by gabrieldonadel** [comment]
> Hi, thanks for upgrading it, do you mind updating the BUCK file for the dependency as well?

Done 😄

**Comment by cortinico** [comment]
#33068 has been marged. You should be able to rebase and get a green CI from that 👍

**Review by cortinico — APPROVED** [review]

**Comment by gabrieldonadel** [comment]
> I think CI is broken after the latest commit, it cannot find some androidx classes:

Yeah, I had quite a hard time installing BUCK in my machine to investigate this because I'm using an M1 mac, I finally got it working, I'll start investigating why this is falling now

**Comment by gabrieldonadel** [comment]
> I think CI is broken after the latest commit, it cannot find some androidx classes:
> 
> ```
> stderr: : warning: unknown enum constant androidx.annotation.RestrictTo.Scope.LIBRARY_GROUP_PREFIX
> 
> /root/react-native/ReactAndroid/src/main/java/com/facebook/react/views/unimplementedview/ReactUnimplementedView.java:22: error: cannot access androidx.core.widget.TintableCompoundDrawablesView
>     mTextView.setLayoutParams(
> ```



Any suggestion on how to fix this @ShikaSD @cortinico ? 

I believe the `cannot access androidx.core.widget.TintableCompoundDrawablesView` error is build caused by some sort of version mismatch in the BUCK file, but I can't quite figure it out

**Comment by cortinico** [comment]
> Any suggestion on how to fix this @ShikaSD @cortinico ?

Yup, you'll need to add to the BUCK file you attached the following rule:
```
fb_native.android_prebuilt_aar(
    name = "appcompat-resources-binary",
    aar = ":appcompat-resources-binary-aar",
)
```

and edit the "appcompat" target with:

```diff
fb_native.android_library(
    name = "appcompat",
    visibility = ["PUBLIC"],
    exported_deps = [
        ":annotation",
        ":appcompat-binary",
+       ":appcompat-resource-binary",
        ":collection",
        ":core",
        ":cursoradapter",
        ":fragment",
        ":legacy-support-core-utils",
        ":vectordrawable",
        ":vectordrawable-animated",
    ],
)
```

If you see the other targets have similar definition

**Comment by gabrieldonadel** [comment]
> > Yup, you'll need to add to the BUCK file you attached the following rule
> 
> Adding that actually ends up breaking another place now 

I managed to fix this by bumping android:core version as well

**Pull request closed without merging**